source/common/grpc/google_async_client_impl.cc

```diff
@@ -90,6 +90,8 @@ GoogleAsyncClientImpl::GoogleAsyncClientImpl(Event::Dispatcher& dispatcher,
   // have comparable overhead to what we are doing in Grpc::AsyncClientImpl, i.e. no expensive
   // new connection implied.
   std::shared_ptr<grpc::Channel> channel = GoogleGrpcUtils::createChannel(config, api);
+  // Get state with try connection = true to ensure the channel is healthy.
+  channel->GetState(/*try_to_connect=*/true);
```

Comment:
I think this `/*try_to_connect=*/` pattern is not used in Envoy. Anyway the comment above suffices.

Reply:
This seems like an unrelated change which should have some kind of independent test. Can we split this out into a different PR?

Reply:
Can I keep it. 
This cache solution is for initialize gRPC channel at channel creation. This GetState(true) is used to poke the gRPC lb at channel creation, it doesn't have any effect no matter it succeeds or fails. But it helps on initialization. Otherwise, the channel establishment still happens at the first request, same as the behavior of existing code.